### PR TITLE
Bumped slick to 3.1.0.

### PR DESCRIPTION
**Pull request by kshakir**
Added Hikari connection pool dependency.
Switched to slick `DatabaseConfig` syntax for instantiating a slick database.

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -3,27 +3,21 @@ database {
   config = hsqldb
 
   hsqldb {
-    url = "jdbc:hsqldb:mem:${slick.uniqueSchema};shutdown=false;hsqldb.tx=mvcc"
-    driver = "org.hsqldb.jdbcDriver"
-    slick.driver = "slick.driver.HsqldbDriver"
+    db.url = "jdbc:hsqldb:mem:memdb;shutdown=false;hsqldb.tx=mvcc"
+    db.driver = "org.hsqldb.jdbcDriver"
```
Comment (mcovarr) [inline src/main/resources/application.conf]: must this prefix be `db`?  it's a bit confusing inside a vendor specific database block and feels specifically like JDBC info, which might be better within a block:

```
...
  jdbc {
    url = "..."
    driver = "..."
  }
...
```

**Comment by mcovarr** [comment]
:+1:

File: src/main/resources/application.conf (hunk 1 above)
Comment (kshakir) [inline src/main/resources/application.conf]: It is JDBC info, mixed in with other slick/hikari specific info. Will block it.

But [slick](http://slick.typesafe.com/doc/3.1.0/database.html?highlight=db#databaseconfig) [wants](https://github.com/slick/slick/blob/3.1.0/slick/src/main/scala/slick/backend/DatabaseConfig.scala#L73) the path named "db", so it shall be.

**Comment by kshakir** [comment]
Rebased. Don't really understand how I took a "hit" on test coverage. Anyways, I think @mcovarr already looked, so @cjllanwarne as second reviewer?

File: src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala [hunk 2]
```diff
@@ -14,9 +16,12 @@ case object ThurloeDatabaseConnector extends DataAccess {
   val configFile = ConfigFactory.load()
 
   // DB Config:
-  val dbConfig = configFile.getConfig("database")
+  val dbConfigName =
+    if (configFile.hasPath("databaseSlickDriverConfigSwitch")) "databaseSlickDriverConfigSwitch" else "database"
```
Comment (cjllanwarne) [inline src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala:20]: This looks like a job for Lenthall!

**Comment by cjllanwarne** [comment]
:+1: 
I think deleting code affects coverage because the proportion of code in the tested/untested categories shifts. Nothing too worrying IMO

File: src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala (hunk 2 above)
Comment (kshakir) [inline src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala:20]: Yeah… currently there is no `getConfigOption`, but maybe in the future.

**Pull request merged**